sc/src/test/java/uk/ac/ebi/atlas/metadata/CellMetadataDaoIT.java

```diff
@@ -0,0 +1,149 @@
+package uk.ac.ebi.atlas.metadata;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.context.web.WebAppConfiguration;
+import uk.ac.ebi.atlas.solr.cloud.SolrCloudCollectionProxyFactory;
+import uk.ac.ebi.atlas.solr.cloud.fullanalytics.SingleCellAnalyticsCollectionProxy;
+import uk.ac.ebi.atlas.testutils.JdbcUtils;
+import uk.ac.ebi.atlas.testutils.RandomDataTestUtils;
+
+import javax.inject.Inject;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ExtendWith(SpringExtension.class)
+@WebAppConfiguration
+@ContextConfiguration(locations = {"classpath:applicationContext.xml", "classpath:dispatcher-servlet.xml"})
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class CellMetadataDaoIT {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CellMetadataDaoIT.class);
+
+    @Inject
+    private SolrCloudCollectionProxyFactory solrCloudCollectionProxyFactory;
+    @Inject
+    private JdbcUtils jdbcUtils;
+
+    private CellMetadataDao subject;
+
+    @BeforeEach
+    void setUp() {
+        this.subject = new CellMetadataDao(solrCloudCollectionProxyFactory);
+    }
+
+    @ParameterizedTest
+    @MethodSource("experimentsWithMetadataProvider")
+    public void validExperimentAccessionHasMetadataFields(String experimentAccession) {
+        assertThat(subject.getMetadataFieldNames(experimentAccession)).isNotEmpty();
+    }
+
+    @Test
+    public void invalidExperimentAccessionHasNoMetadata() {
+        String experimentAccession = RandomDataTestUtils.getRandomExperimentAccession();
+
+        assertThat(subject.getMetadataFieldNames(experimentAccession)).isEmpty();
+    }
+
+    @Test
+    public void invalidCellIdHasNoMetadata() {
+        String experimentAccession = jdbcUtils.fetchRandomSingleCellExperimentAccession();
+        String cellId = "FOOBAR";
+
+        List<SingleCellAnalyticsCollectionProxy.SingleCellAnalyticsSchemaField> metadataFieldNames = subject.getMetadataFieldNames(experimentAccession);
+
+        assertThat(metadataFieldNames).isNotEmpty();
+
+        metadataFieldNames.forEach(field -> {
+            assertThat(subject.getMetadataValueForCellId(experimentAccession, field, cellId)).isNotPresent();
+        });
```

Comment:
In this test and the a few ones below, you can chain both assertions with [`allMatch` or `allSatisfy`](http://joel-costigliola.github.io/assertj/core-8/api/index.html):
```
assertThat(metadataFieldNames)
    .isNotEmpty()
    // Predicate that returns a boolean
    .allMatch(field -> subject.getMetadataValueForCellId(experimentAccession, field, cellId).isEmpty());
```

Alternatively:
```
assertThat(metadataFieldNames)
    .isNotEmpty()
    // Consumer that makes assertions
    .allSatisfy(field -> assertThat(subject.getMetadataValueForCellId(experimentAccession, field, cellId)).isNotPresent());
```